### PR TITLE
[disasters] Date dropdown updates

### PR DESCRIPTION
**Pull request by chejennifer**
- update date dropdown to differentiate between using cache or not
- fix duplicates and missing dates in dropdown

### CODE REVIEW

File: static/js/utils/disaster_event_map_utils.tsx [hunk 1]
```diff
@@ -122,26 +124,28 @@ export function fetchDateList(
...
-      // currDate is in local time but toISOString will get the iso string of the
-      // date in UTC. So first we need to convert currDate to UTC date without
-      // changing the date.
-      const utcDate = new Date(currDate.toLocaleDateString() + " UTC");
-      dateList.push(utcDate.toISOString().substring(0, dateStringCut));
-      if (decrementByYear) {
-        currDate.setFullYear(currDate.getFullYear() - 1);
+    const dateStringCut = increaseByYear ? 4 : 7;
+    const currDate = new Date(minDate + "Z");
+    const endDate = new Date(maxDate + "Z");
```
Comment (beets) [inline static/js/utils/disaster_event_map_utils.tsx:133]: add a note here about the bug?

**Review by beets — APPROVED** [review]
thx for the fix!

File: static/js/utils/disaster_event_map_utils.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/utils/disaster_event_map_utils.tsx:136]: done

**Comment by chejennifer** [comment]
thanks for the review!

**Pull request merged**